### PR TITLE
Add MTW sentence parser

### PR DESCRIPTION
**Pull request by JasminFragnaud**
#54

### CODE REVIEW

File: src/sentences/mtw.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    } else {
+        Ok(do_parse_mtw(sentence.data)?.1)
+    }
+}
+
+fn do_parse_mtw(i: &str) -> IResult<&str, MtwData> {
+    let (i, temperature_value) = opt(float)(i)?;
+    let (i, unit) = opt(preceded(char(','), one_of("C")))(i)?;
+    let unit = unit.map(|ch| match ch {
+        'C' => MtwUnit::Celcius,
```
Comment (elpiel) [inline src/sentences/mtw.rs:64]: Are there any other options apart from `Celsius`? I didn't find anything else, so maybe it's a good idea to just remove the field from the `MtwData`. The documentation can be specific for the field - only Celsius.

File: src/sentences/mtw.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+/// 3:  Mandatory NMEA checksum
+#[derive(Debug, PartialEq)]
+pub struct MtwData {
+    pub temperature: Option<f32>,
+    pub temperature_unit: Option<MtwUnit>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum MtwUnit {
+    Celcius,
```
Comment (elpiel) [inline src/sentences/mtw.rs:31]: There's actually a small typo, should be `Celsius` with an `s`

File: src/sentences/mtw.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+///        |   | |
+/// $--MTW,x.x,C*hh<CR><LF>
+/// ```
+/// 1:  Temperature, degrees
+/// 2:  Unit of Measurement, Celsius
+/// 3:  Mandatory NMEA checksum
+#[derive(Debug, PartialEq)]
+pub struct MtwData {
+    pub temperature: Option<f32>,
+    pub temperature_unit: Option<MtwUnit>,
```
Comment (elpiel) [inline src/sentences/mtw.rs:26]: 1. Let's just leave `temperature` and add a documentation block `///` that it is only "degrees Celsius"
2. use `double` (f64) instead. We've discussed this before and we should unify the api to always use `f64`.

File: src/sentences/mtw.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    #[test]
+    fn test_parse_mtw_invalid_unit() {
+        let s = parse_nmea_sentence("$INMTW,17.9,x*20").unwrap();
+        assert_eq!(s.checksum, s.calc_checksum());
+        assert_eq!(s.checksum, 0x20);
+        let mtw_data = parse_mtw(s).unwrap();
+        assert_eq!(Some(17.9), mtw_data.temperature);
+        assert_eq!(None, mtw_data.temperature_unit);
```
Comment (elpiel) [inline src/sentences/mtw.rs:99]: This test should cause an error instead of `Option::None`

**Review by elpiel — CHANGES_REQUESTED** [review]
Thank you very much for your fist-time contribution :tada: 

I have left a few notes on the PR, I'm eager to merge it asap :blush:

File: src/sentences/mtw.rs (hunk 1 above)
Comment (JasminFragnaud) [reply src/sentences/mtw.rs:64]: I've had searched for this but it's kinda strange, it's described as a field in the documentation (and implemented as is in other libs) but this field seems to can only be celsius. Also didn't find any example with something else, so i assume we can remove it.

Comment (elpiel) [reply src/sentences/mtw.rs:64]: Awesome!

**Review by elpiel — APPROVED** [review]
LGTM! Thank you again for your contribution

**Pull request merged**